ui/src/app/utils/runtime-utils.tsx

```diff
@@ -334,10 +375,14 @@ export const useRuntimeStatus = (currentWorkspaceNamespace, currentGoogleProject
     initializePolling();
   }, [runtimeStatus]);
 
-  const setStatusRequest = async(req) => {
+  const setStatusRequest = async(req, deletePD) => {
     await switchCase(req,
       [RuntimeStatusRequest.Delete, () => {
-        return runtimeApi().deleteRuntime(currentWorkspaceNamespace);
+        if (enablePD) {
```

Comment:
This isn't really necessary here I don't think - leave feature flagging to the caller